### PR TITLE
Skip CI in some cases to save resources

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>. NO
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md). NO NEED

We have conditional execution for plugin tests, but for E2E tests, we want to make them "required" for status check while saving resources for some changes like "\*.md" and "skywalking-ui/*", this patch check the changed files of a pull request and set `SKIP_CI` to skip the workflows.

### CODE REVIEW

File: .github/actions/skip/action.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#
+name: "Set Skip Env Var"
+description: "Action to set the SKIP_CI environment variable indicating that we should skip CI jobs"
+inputs:
+  paths:
+    description: >-
+      Set the SKIP_CI environment variable when and only when all the changed files located in one of the path,
+      the paths is shell-style pattern.
+    required: false
+    default: >-
```
Comment (wu-sheng) [inline .github/actions/skip/action.yml:27]: So md and ui submodule changes wouldn't trigger any CI and e2e tasks anymore, right? Do we need to set more required tasks?

Comment (kezhenxu94) [reply .github/actions/skip/action.yml:27]: > So md and ui submodule changes wouldn't trigger any CI and e2e tasks anymore, right?

Yes, effectively, no CI task will be run when only `*.md` and `skywalking-ui` are updated.

> Do we need to set more required tasks?

We can set all E2E tests as required, WDYT?

File: test/e2e/e2e-service-provider/pom.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
     <artifactId>e2e-service-provider</artifactId>
 
     <properties>
-        <sw.snapshot.version>8.4.0-SNAPSHOT</sw.snapshot.version>
+        <sw.snapshot.version>8.4.0</sw.snapshot.version>
```
Comment (wu-sheng) [inline test/e2e/e2e-service-provider/pom.xml:37]: The variable name is not correct.

File: .github/actions/skip/action.yml (hunk 1 above)
Comment (wu-sheng) [reply .github/actions/skip/action.yml:27]: > We can set all E2E tests as required, WDYT?

I want to, but meter system tests are not stable from time to time. I am not sure why.

Comment (kezhenxu94) [reply .github/actions/skip/action.yml:27]: > > We can set all E2E tests as required, WDYT?
> 
> I want to, but meter system tests are not stable from time to time. I am not sure why.

Well, let's keep it as is for now

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**